client/example/main.go

```diff
@@ -0,0 +1,46 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	rest "github.com/SKF/go-rest-utility/client"
+	"github.com/SKF/go-rest-utility/client/auth"
+)
+
+type GetNodeResponse struct {
+	Node Node `json:"node"`
+}
+
+type Node struct {
+	Label       string `json:"label"`
+	Description string `json:"description"`
+}
+
+func main() {
+	identityToken := "eyj..."
+
+	ctx := context.Background()
+
+	client := rest.NewClient(
+		rest.WithBaseURL("https://api.sandbox.hierarchy.enlight.skf.com/"),
+		rest.WithTokenProvider(auth.IdentityToken(identityToken)),
+	)
+
+	request := rest.Get("nodes/{id}").
+		Assign("id", "df3214a6-2db7-11e8-b467-0ed5f89f718b").
+		SetHeader("Accept", "application/json")
+
+	response, err := client.Do(ctx, request)
```

Comment:
It might be nice with a DoAndUnmarshal function?

maybe something like err := client.DoAndUnmarshal(ctx, request, resObject)

I'm really not sure on this one.

Reply:
Added it, as you said. It might be nice.